src/ray/gcs/redis_context.cc

```diff
@@ -630,15 +662,40 @@ std::unique_ptr<CallbackReply> RedisContext::RunArgvSync(
     argv.push_back(arg.data());
     argc.push_back(arg.size());
   }
-  auto redis_reply = reinterpret_cast<redisReply *>(
-      ::redisCommandArgv(context_.get(), args.size(), argv.data(), argc.data()));
-  if (redis_reply == nullptr) {
-    RAY_LOG(ERROR) << "Failed to send redis command (sync): " << context_->errstr;
-    return nullptr;
+  // Run the command with exponential backoff retries.
```

Comment:
you also added exponential backoff, SYNC_ONLY and other nuances. Let's document it well on our behavior